evidence/pool.go

```diff
@@ -254,35 +255,51 @@ func (evpool *Pool) fastCheck(ev types.Evidence) bool {
 			return false
 		}
 		if err != nil {
-			evpool.logger.Error("Failed to load light client attack evidence", "err", err, "height", ev.Height(), "key", key)
+			evpool.logger.Error("Failed to load light client attack evidence", "err", err, "key(height/hash)", key)
 			return false
 		}
 		var trustedPb tmproto.LightClientAttackEvidence
 		err = trustedPb.Unmarshal(evBytes)
 		if err != nil {
 			evpool.logger.Error("Failed to convert light client attack evidence from bytes",
-				"err", err, "height", ev.Height(), "key", key)
+				"err", err, "key(height/hash)", key)
 			return false
 		}
 		trustedEv, err := types.LightClientAttackEvidenceFromProto(&trustedPb)
 		if err != nil {
 			evpool.logger.Error("Failed to convert light client attack evidence from protobuf",
-				"err", err, "height", ev.Height(), "key", key)
+				"err", err, "key(height/hash)", key)
 			return false
 		}
 		// ensure that all the byzantine validators that the evidence pool has match the byzantine validators
 		// in this evidence
 		if len(trustedEv.ByzantineValidators) != len(lcae.ByzantineValidators) {
 			return false
 		}
-		for idx, val := range trustedEv.ByzantineValidators {
-			if !bytes.Equal(lcae.ByzantineValidators[idx].Address, val.Address) {
+
+		byzValsCopy := make([]*types.Validator, len(lcae.ByzantineValidators))
+		for i, v := range lcae.ByzantineValidators {
+			byzValsCopy[i] = v.Copy()
+		}
+
+		trustedValsCopy := make([]*types.Validator, len(trustedEv.ByzantineValidators))
+		for i, v := range trustedEv.ByzantineValidators {
+			trustedValsCopy[i] = v.Copy()
+		}
+
+		// ensure that both validator arrays are in the same order
+		sort.Sort(types.ValidatorsByVotingPower(byzValsCopy))
+		sort.Sort(types.ValidatorsByVotingPower(trustedValsCopy))
```

Comment:
aren't trustedVals sorted already? we should sort them when we add the evidence to a database.

Reply:
I didn't want to mutate the evidence, but I guess we can. We can just assert an ordering style i.e. ByzantineValidators have to be ordered based on power regardless of how the validator sets are ordered

Reply:
Well, we are not really changing it per se, just enforcing order (similarly to sorting votes in DuplicateVoteEvidence).